src/libs/NetworkConnection.js

```diff
@@ -53,16 +53,18 @@ function setOfflineStatus(isCurrentlyOffline) {
  */
 function subscribeToNetInfo() {
     // Calling NetInfo.configure (re)checks current state. We use it to force a recheck whenever we (re)subscribe
-    NetInfo.configure({
-        // By default, for web (including Electron) NetInfo uses `/` for `reachabilityUrl`
-        // When App is served locally or from Electron this would respond with OK even with no internet
-        // Using API url ensures reachability is tested over internet
-        reachabilityUrl: CONFIG.EXPENSIFY.URL_API_ROOT,
-        reachabilityTest: response => Promise.resolve(response.status === 200),
-
-        // If a check is taking longer than this time we're considered offline
-        reachabilityRequestTimeout: CONST.NETWORK.MAX_PENDING_TIME_MS,
-    });
+    if (CONFIG.IS_IN_PRODUCTION) {
+        NetInfo.configure({
+            // By default, for web (including Electron) NetInfo uses `/` for `reachabilityUrl`
+            // When App is served locally or from Electron this would respond with OK even with no internet
+            // Using API url ensures reachability is tested over internet
+            reachabilityUrl: CONFIG.EXPENSIFY.URL_API_ROOT,
+            reachabilityTest: response => Promise.resolve(response.status === 200),
+
+            // If a check is taking longer than this time we're considered offline
+            reachabilityRequestTimeout: CONST.NETWORK.MAX_PENDING_TIME_MS,
+        });
+    }
```

Comment:
Even though we're only calling `NetInfo.configure(config)` in PROD, in DEV it's still implicitly uses a default configuration

Which mean that NetInfo might be acting differently in DEV

-  `reachabilityUrl` is `/` this is the local dev server, even if you turn off your internet connection your local dev server is reachable tricking NetInfo into thinking it's online 

- `reachabilityRequestTimeout` for the default configuration is different than what we have on PROD

Maybe this won't be a problem for local development - you can simulate you are offline by tweaking network settings in chrome, but if you actually don't have internet (or simulate it by turning it off on the router) it won't be detected (at least for external contributors) (web|desktop)

Reply:
Yes, sounds good. If there's a better solution feel free to push it up please. I needed to quickly get this working to avoid further confusion within the team.

Reply:
Would this work: https://github.com/Expensify/App/pull/8049
```js
reachabilityUrl: `${CONFIG.EXPENSIFY.URL_API_ROOT}api`,
```

It works for me as external contributor, I think it should work with ngrok too, but unsure how to test it locally

Reply:
Great idea, let me try it!

Reply:
Ah it does work and makes sense because we set the allow origin headers in `/api`. Amazing !